src/Microsoft.Framework.Runtime/Servicing/ServicingBreadcrumbs.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet;
+
+namespace Microsoft.Framework.Runtime.Servicing
+{
+    internal class ServicingBreadcrumbs
+    {
+        private readonly string _breadcrumbsFolder;
+
+        public ServicingBreadcrumbs()
+            : this(ResolveBreadcrumbsFolder())
+        {
+        }
+
+        internal ServicingBreadcrumbs(string breadcrumbsFolder)
+        {
+            // If the directory doesn't exist, don't create it because it
+            // needs special permissions on it
+            _breadcrumbsFolder = Directory.Exists(breadcrumbsFolder) ? breadcrumbsFolder : null; ;
+        }
+
+        public static bool IsPackageServiceable(IPackage package)
+        {
+            // TODO: Figure out what makes a package serviceable
+            return true;
+        }
+
+        public void LeaveBreadcrumb(string packageId, SemanticVersion packageVersion)
+        {
+            if (string.IsNullOrWhiteSpace(_breadcrumbsFolder))
+            {
+                return;
+            }
+
+            // Create both files for now until we get clear instructions about the format of the name
+            CreateBreadcrumbfile(packageId);
+            CreateBreadcrumbfile(packageId + "." + packageVersion);
+        }
+
+        private static string ResolveBreadcrumbsFolder()
+        {
+            var programDataFolder = Environment.GetEnvironmentVariable("ProgramData");
```

Comment:
Consider using SpecialFolder.CommonApplicationData instead so that we're not hardcoding this.